compiler/noirc_frontend/src/hir/aztec_library.rs

```diff
@@ -209,19 +203,30 @@ fn include_relevant_imports(ast: &mut SortedModule) {
 }
 
 /// Creates an error alerting the user that they have not downloaded the Aztec-noir library
-fn check_for_aztec_dependency(crate_id: &CrateId, context: &Context) -> Result<(), FileId> {
+fn check_for_aztec_dependency(
+    crate_id: &CrateId,
+    context: &Context,
+) -> Result<(), (DefCollectorErrorKind, FileId)> {
     let crate_graph = &context.crate_graph[crate_id];
     let has_aztec_dependency = crate_graph.dependencies.iter().any(|dep| dep.as_name() == "aztec");
     if has_aztec_dependency {
         Ok(())
     } else {
-        Err(crate_graph.root_file_id)
+        Err((DefCollectorErrorKind::AztecNotFound {}, crate_graph.root_file_id))
     }
 }
 
 // Check to see if the user has defined a storage struct
 fn check_for_storage_definition(module: &SortedModule) -> bool {
-    module.types.iter().any(|function| function.name.0.contents == "Storage")
+    module.types.iter().any(|r#struct| r#struct.name.0.contents == "Storage")
+}
+
+// Check to see if the user has defined a compute_note_hash_and_nullifier function exists
+fn check_for_compute_note_hash_and_nullifier_definition(module: &SortedModule) -> bool {
+    module
+        .functions
+        .iter()
+        .any(|func| func.def.name.0.contents == "compute_note_hash_and_nullifier")
```

Comment:
now i think of it, i think that compute note hash and nullifier will also cause errors if the function selector is not identical to the one specified in the docs. 
I do not want to block this pr, but we probably need:
    unconstrained fn compute_note_hash_and_nullifier(contract_address: Field, nonce: Field, storage_slot: Field, preimage: [Field; TOKEN_NOTE_LEN]) -> [Field; 4] 

to check for the types